docs: Move Files and update TOC

Balaji2198 opened the pull request:
### Files Deleted 
* `/docs/brief-install-vagrant-dev.md`

### Files Moved
* `/docs/code-of-conduct.md` -> `/docs/contributing/code-of-conduct.md`
*  `/docs/migration-renumbering.md` -> `/docs/subsystems/migration-renumbering.md`
* `/docs/users.md` -> `/docs/subsystems/users.md`

### TOC Update
* `/docs/contributing/index.rst`
* `/docs/production/index.rst`
* `/docs/subsystems/index.rst`

Fixes #7417  First Two Parts

Balaji2198 commented:
@timabbott  Kindly Review it.

timabbott commented:
@Balaji2198 thanks for working on this!  A few requests:
* I suspect this will break some links.  You can check for this with `git grep input-pills` and `./tools/test-documentation` (do both for each article you rename, since `test-documentation` only checks links in the docs/ files, but we might have some from code as well).
* Can you do this as a series of smaller commits that each delete/move one article?  That's a lot easier to review.  Check out our [GitHub guide](http://zulip.readthedocs.io/en/latest/contributing/git-guide.html) and search for "rebase" for advice on how to do this as rebase.  Our [commit guidelines](http://zulip.readthedocs.io/en/latest/version-control.html) are also helpful reading.

Balaji2198 commented:
@timabbott  I rebased the commit and separated the commits one per file. 
And did a `git grep`, Fixed the links shown. 
Also added the title of `users.md`. (Because To include in TOC).
Your thoughts on it? Don't know why Travis Fails.

timabbott commented:
The test failures are because you didn't move everything (e.g. the `backend` / linter failures are because a linter exclude rule didn't get updated)
```
(zulip-py3-venv) tabbott@zaset:~/zulip$ git grep migration-renumbering
docs/subsystems/index.rst:   migration-renumbering
docs/subsystems/schema-migrations.md:  manually. There is a tutorial [here](migration-renumbering.html) that
tools/linter_lib/custom_check.py:            "docs/migration-renumbering.md",
```

How are you testing this?  When I ran `tools/build-docs`, I got this error:
```
/home/tabbott/zulip/docs/contributing/code-of-conduct.rst: WARNING: document not readable. Ignored.
```

which was because you needed to update the `code-of-conduct.md` symlink after moving it.

timabbott commented:
Since I think it'll help you learn, I ended up cleaning up most of these commits and merging them.  @Balaji2198 can you look at the 5 commits ending with 489feaf13936092436d0736a2ff800643a0eec5b.  Notice how we do changes atomically together, and explain why each change is correct or desirable in the commit message.

timabbott commented:
Thanks for your work on this @Balaji2198!

Balaji2198 commented:
@timabbott  Now I understood. Thanks. :+1:
